sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/higherOrderFunctions.scala

```diff
@@ -130,6 +134,23 @@ case class LambdaFunction(
 
   override def eval(input: InternalRow): Any = function.eval(input)
 
+  override def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
+    val functionCode = function.genCode(ctx)
```

Comment:
seems we can just return `function.genCode(ctx)`?

Reply:
The code here is only to optimize the non-nullable case, which should be handled by the function body codegen already.

Reply:
I'm trying to wrap my ahead around this again. I _think_ they might be doing different things? Like for `ArrayTransform`, the `nullSafeCodeGen` is only optimizing the non-nullable case of the result of `ArrayTransform`, whereas this is optimizing the non-nullable case of the result of the lambda function being called for each element in the `ArrayTransform`

Reply:
But maybe this already handles that:
```
        val resultNull = if (function.nullable) Some(functionCode.isNull.toString) else None
        val resultAssignment = CodeGenerator.setArrayElement(arrayData, dataType.elementType,
          i, copy, isNull = resultNull)
```

I'll need to think about this a little more

Reply:
Okay nevermind I see what you mean now, it's just assigning the same values to new variables for no reason. Changed to just return `function.genCode(ctx)`